From ff1aeca63987442509b93a2d714c418acd827162 Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 13 Jun 2014 14:35:30 -0400
Subject: [block] blk-mq: free hctx->ctx_map when init failed

Message-id: <1402670238-13916-43-git-send-email-snitzer@redhat.com>
Patchwork-id: 83802
O-Subject: [RHEL7.1 PATCH 042/150] blk-mq: free hctx->ctx_map when init failed
Bugzilla: 1105204
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Jeff Moyer <jmoyer@redhat.com>

BZ: 1105204

Upstream commit 11471e0d04f3762b9216d8421ce6e9d89b0bf450
Author: Ming Lei <tom.leiming@gmail.com>
Date:   Sat Apr 19 18:00:16 2014 +0800

    blk-mq: free hctx->ctx_map when init failed

    Avoid memory leak in the failure path.

    Reviewed-by: Christoph Hellwig <hch@lst.de>
    Signed-off-by: Ming Lei <tom.leiming@gmail.com>
    Signed-off-by: Jens Axboe <axboe@fb.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/block/blk-mq.c b/block/blk-mq.c
index ee225cc..5fbbb22 100644
--- a/block/blk-mq.c
+++ b/block/blk-mq.c
@@ -1289,6 +1289,7 @@ static int blk_mq_init_hw_queues(struct request_queue *q,
 
   blk_mq_unregister_cpu_notifier(&hctx->cpu_notifier);
   kfree(hctx->ctxs);
+  kfree(hctx->ctx_map);
  }
 
  return 1;
-- 
1.7.1